octobot_trading/personal_data/positions/position.py

```diff
@@ -305,7 +311,12 @@ def update_from_order(self, order):
 
         # Updates position average entry price from order only when increasing position side
         if self._is_update_increasing_size(size_update):
+            if self.size == constants.ZERO:
+                self.first_entry_time = self.exchange_manager.exchange.get_exchange_current_time()
             self.update_average_entry_price(size_update, order.filled_price)
+        elif self._is_update_decreasing_size(size_update):
+            self.update_average_exit_price(size_update, order.filled_price)
+            self.already_reduced_size += size_update
```

Comment:
A there we should have something like `_update_exit_price(size_update, on_liquidation=False)` (previous comment)